src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -334,7 +314,7 @@ protected virtual Task ApplyResponseGrantAsync()
 
         public virtual void SignIn(ISignInContext context)
         {
-            SignInIdentityContext = new SignInIdentityContext(context.Principal, new AuthenticationProperties(context.Properties));
+            SignInContext = new SignInContext(context.Principal, new AuthenticationProperties(context.Properties));
```

Comment:
Added tests verifying that exceptions are thrown (rather than sign in/sign out happening) when wrong auth type is specified